openverse_catalog/dags/providers/provider_api_scripts/wordpress.py

```diff
@@ -13,284 +13,185 @@
+            "filetype": filetype,
+            "filesize": filesize,
+            "license_info": self.license_info,
+            "meta_data": metadata,
+            "raw_tags": tags,
+        }
+
+    def _get_file_info(self, media_details):
+        preferred_sizes = ["2048x2048", "1536x1536", "medium_large", "large", "full"]
+        for size in preferred_sizes:
+            file_details = media_details.get("sizes", {}).get(size, {})
+            image_url = file_details.get("source_url")
+            if not image_url or image_url == "":
+                continue
+
+            height = file_details.get("height")
+            width = file_details.get("width")
+            filetype = None
+            if filename := file_details.get("file"):
+                filetype = Path(filename).suffix.replace(".", "")
+
+            filesize = (
+                media_details.get("filesize", 0)
+                if size == "full"
+                else file_details.get("filesize", 0)
+            )
+            if not filesize or int(filesize) == 0:
+                filesize = self._get_filesize(image_url)
+
+            return image_url, height, width, filetype, filesize
+        return None, None, None, None, None
+
+    def _get_filesize(self, image_url):
+        resp = self.get_response_json(query_params={}, endpoint=image_url)
+        if resp:
+            filesize = int(resp.headers.get("Content-Length", 0))
+            return filesize if filesize != 0 else None
+
+    @staticmethod
+    def _get_author_data(image):
+        try:
+            raw_author = image.get("_embedded", {}).get("author", [])[0]
+        except IndexError:
+            return None, None
+        author = raw_author.get("name")
+        if author is None or author == "":
+            author = raw_author.get("slug")
+        author_url = raw_author.get("url")
+        if author_url == "":
+            author_url = raw_author.get("link")
+        return author, author_url
+
+    @staticmethod
+    def _get_title(image):
+        if title := image.get("content", {}).get("rendered"):
+            try:
+                title = html.fromstring(title).text_content()
+            except UnicodeDecodeError as e:
+                logger.warning(f"Can't save the image's title ('{title}') due to {e}")
+                return None
+        return title
+
+    @staticmethod
+    def _get_metadata(media_data, media_details):
+        raw_metadata = media_details.get("image_meta", {})
+        metadata, tags = {}, []
+        extras = [
+            "aperture",
+            "camera",
+            "created_timestamp",
+            "focal_length",
+            "iso",
+            "shutter_speed",
+        ]
+        for key in extras:
+            value = raw_metadata.get(key)
+            if value not in [None, ""]:
+                metadata[key] = value
+
+        raw_related_resources = media_data.get("_embedded", {}).get("wp:term", [])
+        resource_mapping = {
```

Comment:
Not for this PR, but in general, would there be a lot of memory/computational savings if we declared the `resource_mappin` and `extras` (from line 157) in one place, instead of every time we call `_get_metadata` here. And if it would be beneficial, where would we save it so that it's not created every time DAGs are parsed?

Reply:
That's a good call! I think defining a dict like this on the class is OK, as long as it's not making a DB or API call at parse time 😄